### ISSUE TITLE
Cmake build on Windows fails when compiling against the Ninja generator (targeting x64 msvc)

### ISSUE THREAD

**Issue by jeremyong** [microsoft/msquic#1401]
Use the Ninja generator (default generator used by Visual Studio's cmake tools) and target x64 (note that this happens implicitly based on machine architecture as Ninja does not accept the `-A` argument).

The first error is that `QUIC_TL` set as `openssl` attempts to determine the architecture based on `CMAKE_GENERATOR_PLATFORM` instead of `CMAKE_SYSTEM_PROCESSOR` which is available regardless of generator selected and host/target (when cross compiling). 

I can get around this by making the following change:

```cmake
        string(TOLOWER ${CMAKE_SYSTEM_PROCESSOR} system_processor)

        if (${system_processor} STREQUAL "aarch64")
            set(QUIC_OPENSSL_WIN_ARCH "VC-WIN64-ARM")
        elseif (${system_processor} STREQUAL "arm" OR ${system_processor} STREQUAL "armv71")
            set(QUIC_OPENSSL_WIN_ARCH "VC-WIN32-ARM")
        elseif (${system_processor} STREQUAL "i686")
            set(QUIC_OPENSSL_WIN_ARCH "VC-WIN32")
        elseif (${system_processor} STREQUAL "x86_64" OR ${system_processor} STREQUAL "amd64")
            set(QUIC_OPENSSL_WIN_ARCH "VC-WIN64A")
        else()
            message(FATAL_ERROR "Unknown system processor ${system_processor}")
        endif()
```

The next thing that breaks is the way imported targets are injected into the dependency graph. The `OpenSSL_Build` custom target doesn't explicitly state that it will produce `libssl.lib` and `libcrypto.lib` as a byproduct, and so Ninja will not realize this target needs to be built (it will fail to build `msquic.dll` because it can't deduce that any known target will produce the ssl libs needed). To fix this, the `BYPRODUCTS` argument should be supplied to `OpenSSL_Build` as below:

```cmake
            if(${CMAKE_BUILD_TYPE} STREQUAL "Debug")
               set(libssl_out "${OPENSSL_DIR}/debug/lib/libssl${CMAKE_STATIC_LIBRARY_SUFFIX}")
               set(libcrypto_out "${OPENSSL_DIR}/debug/lib/libcrypto${CMAKE_STATIC_LIBRARY_SUFFIX}")
           else()
               set(libssl_out "${OPENSSL_DIR}/release/lib/libssl${CMAKE_STATIC_LIBRARY_SUFFIX}")
               set(libcrypto_out "${OPENSSL_DIR}/release/lib/libcrypto${CMAKE_STATIC_LIBRARY_SUFFIX}")
           endif()


            add_custom_target(OpenSSL_Build
                DEPENDS $<IF:$<CONFIG:Debug>,${QUIC_BUILD_DIR}/submodules/openssl/debug/makefile,${QUIC_BUILD_DIR}/submodules/openssl/release/makefile>
                WORKING_DIRECTORY $<IF:$<CONFIG:Debug>,${QUIC_BUILD_DIR}/submodules/openssl/debug,${QUIC_BUILD_DIR}/submodules/openssl/release>
                BYPRODUCTS ${libssl_out} ${libcrypto_out}
                COMMAND nmake install_dev)
            set_property(TARGET OpenSSL_Build PROPERTY FOLDER "helpers")
```

Note that I don't believe `BYPRODUCTS` supports generator expressions, hence the helper variables above for `libssl_out` and `libcrypto_out`.

The last issue I encounter is a series of warnings emitted in `intrin0.h` which end up being treated as errors:

```
 [170/359] Building C object src\core\CMakeFiles\core.dir\congestion_control.c.obj
  FAILED: src/core/CMakeFiles/core.dir/congestion_control.c.obj 
  C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1428~1.299\bin\Hostx64\x64\cl.exe  /nologo -DQUIC_DISABLE_CLIENT_CERT_TESTS -DQUIC_EVENTS_MANIFEST_ETW -DQUIC_LOGS_MANIFEST_ETW -DSECURITY_WIN32 -DVER_BUILD_ID=0 -DVER_SUFFIX=-private -DWIN32_LEAN_AND_MEAN -Iclog -Iinc\core.clog -I..\..\..\src\inc -Iinc /DWIN32 /D_WINDOWS /Zi /Ob0 /Od /RTC1 -MTd /analyze /MP /WX /W4 /sdl -std:c11 /showIncludes /Fosrc\core\CMakeFiles\core.dir\congestion_control.c.obj /Fdsrc\core\CMakeFiles\core.dir\core.pdb /FS -c ..\..\..\src\core\congestion_control.c
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(122): error C2220: the following warning is treated as an error
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(122): warning C28251: Inconsistent annotation for '_BitScanForward': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(123): warning C28251: Inconsistent annotation for '_BitScanForward64': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(125): warning C28251: Inconsistent annotation for '_BitScanReverse': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(126): warning C28251: Inconsistent annotation for '_BitScanReverse64': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(128): warning C28251: Inconsistent annotation for '_bittest': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(131): warning C28251: Inconsistent annotation for '_InterlockedAnd': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(132): warning C28251: Inconsistent annotation for '_InterlockedAnd16': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(136): warning C28251: Inconsistent annotation for '_InterlockedAnd64': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(140): warning C28251: Inconsistent annotation for '_InterlockedAnd8': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(147): warning C28251: Inconsistent annotation for '_InterlockedCompareExchange': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(149): warning C28251: Inconsistent annotation for '_InterlockedCompareExchange16': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(153): warning C28251: Inconsistent annotation for '_InterlockedCompareExchange64': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(164): warning C28251: Inconsistent annotation for '_InterlockedCompareExchange128': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(168): warning C28251: Inconsistent annotation for '_InterlockedDecrement': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(170): warning C28251: Inconsistent annotation for '_InterlockedDecrement16': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(171): warning C28251: Inconsistent annotation for '_InterlockedDecrement64': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(172): warning C28251: Inconsistent annotation for '_InterlockedExchange': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(174): warning C28251: Inconsistent annotation for '_InterlockedExchange16': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(178): warning C28251: Inconsistent annotation for '_InterlockedExchange64': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(182): warning C28251: Inconsistent annotation for '_InterlockedExchange8': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(186): warning C28251: Inconsistent annotation for '_InterlockedExchangeAdd': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(191): warning C28251: Inconsistent annotation for '_InterlockedExchangeAdd64': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(195): warning C28251: Inconsistent annotation for '_InterlockedExchangeAdd8': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(205): warning C28251: Inconsistent annotation for '_InterlockedIncrement': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(207): warning C28251: Inconsistent annotation for '_InterlockedIncrement16': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(208): warning C28251: Inconsistent annotation for '_InterlockedIncrement64': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(210): warning C28251: Inconsistent annotation for '_InterlockedOr': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(211): warning C28251: Inconsistent annotation for '_InterlockedOr16': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(215): warning C28251: Inconsistent annotation for '_InterlockedOr64': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(219): warning C28251: Inconsistent annotation for '_InterlockedOr8': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(226): warning C28251: Inconsistent annotation for '_InterlockedXor': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(227): warning C28251: Inconsistent annotation for '_InterlockedXor16': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(231): warning C28251: Inconsistent annotation for '_InterlockedXor64': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(235): warning C28251: Inconsistent annotation for '_InterlockedXor8': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(253): warning C28251: Inconsistent annotation for '_interlockedbittestandset': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(262): warning C28251: Inconsistent annotation for '__popcnt64': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(263): warning C28251: Inconsistent annotation for '__shiftright128': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin0.h(268): warning C28251: Inconsistent annotation for '_umul128': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\xmmintrin.h(342): warning C28251: Inconsistent annotation for '_mm_prefetch': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\xmmintrin.h(351): warning C28251: Inconsistent annotation for '_mm_setcsr': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\emmintrin.h(399): warning C28251: Inconsistent annotation for '_mm_clflush': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(86): warning C28251: Inconsistent annotation for '_InterlockedCompareExchangePointer': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(101): warning C28251: Inconsistent annotation for '_InterlockedExchangePointer': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(148): warning C28251: Inconsistent annotation for '__addgsbyte': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(149): warning C28251: Inconsistent annotation for '__addgsdword': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(150): warning C28251: Inconsistent annotation for '__addgsqword': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(151): warning C28251: Inconsistent annotation for '__addgsword': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(164): warning C28251: Inconsistent annotation for '__fastfail': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(175): warning C28251: Inconsistent annotation for '__incgsbyte': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(176): warning C28251: Inconsistent annotation for '__incgsdword': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(177): warning C28251: Inconsistent annotation for '__incgsqword': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(178): warning C28251: Inconsistent annotation for '__incgsword': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(192): warning C28251: Inconsistent annotation for '__movsb': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(193): warning C28251: Inconsistent annotation for '__movsd': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(194): warning C28251: Inconsistent annotation for '__movsq': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(195): warning C28251: Inconsistent annotation for '__movsw': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(196): warning C28251: Inconsistent annotation for '__mulh': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(230): warning C28251: Inconsistent annotation for '__readgsbyte': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(231): warning C28251: Inconsistent annotation for '__readgsdword': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(232): warning C28251: Inconsistent annotation for '__readgsqword': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(233): warning C28251: Inconsistent annotation for '__readgsword': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(235): warning C28251: Inconsistent annotation for '__readpmc': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(240): warning C28251: Inconsistent annotation for '__segmentlimit': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(242): warning C28251: Inconsistent annotation for '__shiftleft128': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(245): warning C28251: Inconsistent annotation for '__stosb': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(246): warning C28251: Inconsistent annotation for '__stosd': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(247): warning C28251: Inconsistent annotation for '__stosq': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(248): warning C28251: Inconsistent annotation for '__stosw': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(263): warning C28251: Inconsistent annotation for '__umulh': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(294): warning C28251: Inconsistent annotation for '__writegsbyte': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(295): warning C28251: Inconsistent annotation for '__writegsdword': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(296): warning C28251: Inconsistent annotation for '__writegsqword': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(297): warning C28251: Inconsistent annotation for '__writegsword': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(303): warning C28251: Inconsistent annotation for '_bittest64': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(304): warning C28251: Inconsistent annotation for '_bittestandcomplement': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(305): warning C28251: Inconsistent annotation for '_bittestandcomplement64': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(306): warning C28251: Inconsistent annotation for '_bittestandreset': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(307): warning C28251: Inconsistent annotation for '_bittestandreset64': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(308): warning C28251: Inconsistent annotation for '_bittestandset': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(309): warning C28251: Inconsistent annotation for '_bittestandset64': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(315): warning C28251: Inconsistent annotation for '_interlockedbittestandreset': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(316): warning C28251: Inconsistent annotation for '_interlockedbittestandreset64': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(323): warning C28251: Inconsistent annotation for '_interlockedbittestandset64': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(400): warning C28251: Inconsistent annotation for '_m_prefetchw': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(473): warning C28251: Inconsistent annotation for '_mm_clflush': this instance has no annotations. See <no file>(0). Note: There are several prototypes for this function. This warning compares the first with instance number 3.
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(740): warning C28251: Inconsistent annotation for '_mm_prefetch': this instance has no annotations. See <no file>(0). Note: There are several prototypes for this function. This warning compares the first with instance number 3.
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(772): warning C28251: Inconsistent annotation for '_mm_setcsr': this instance has no annotations. See <no file>(0). Note: There are several prototypes for this function. This warning compares the first with instance number 3.
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(897): warning C28251: Inconsistent annotation for '_mul128': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(899): warning C28251: Inconsistent annotation for '_rotl16': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(901): warning C28251: Inconsistent annotation for '_rotl8': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(903): warning C28251: Inconsistent annotation for '_rotr16': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(905): warning C28251: Inconsistent annotation for '_rotr8': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\intrin.h(906): warning C28251: Inconsistent annotation for '_setjmp': this instance has no annotations. See <no file>(0).
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um\ws2tcpip.h(968): warning C6101: Returning uninitialized memory '*Mtu'.  A successful path through the function does not set the named _Out_ parameter. : Lines: 968, 973, 974, 975, 976, 978, 979, 994, 968
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\shared\mstcpip.h(1031): warning C6385: Reading invalid data from 'a->u.Byte':  the readable size is '16' bytes, but '17' bytes may be read.: Lines: 1027, 1028, 1029, 1031
C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\shared\mstcpip.h(1031): warning C6385: Reading invalid data from 'b->u.Byte':  the readable size is '16' bytes, but '17' bytes may be read.: Lines: 1027, 1028, 1029, 1031
```

This happens in a number of files that transitively include this header. I've checked that all the linker flags correctly pass `machine:x64` as assigned by cmake for the target architecture and other 64-bit programs compile and can include these headers/intrinsics without issue. In particular, the compiler sets `_M_X64` as one would expect. I can only surmise that for whatever reason, the incorrect CRT is being linked but I'm not sure exactly where/how to diagnose this as I haven't encountered this issue before.

**Comment by ThadHouse** [comment]
I've looked into doing this before, and ran into similar issues that I did not have time to investigate. We would happily accept a PR to fix these issues and get Ninja working on Windows, but otherwise this issue will be placed in the backlog if we have extra time. The last error seems to be caused by `/analyze`, so its possible that flag is just incompatible with Ninja and would need to be removed in Ninja builds.
